src/export.c

```diff
@@ -129,8 +129,6 @@ bool export_wav(MusSong *song, CydWavetableEntry * entry, FILE *f, int channel)
 		}
 	}
 	
-	return true; // Successful (not aborted)

```

Comment:
I think this is the main issue here: it returns before deinit (also the cyd instance isn't deinit and ww_finish doesn't get to free the memory). Perhaps instead of the revert we could just set a success flag here and return that very last.

Reply:
Or, just make sure ww_finish is called the return (be it a success or a failure)

Reply:
If we really want `export_wav` to continue to return a success boolean, having a success flag and returning that at the end of the function is the way to go.
However, I only made `export_wav` return a boolean to indicate whether `fclose` still needs to be called. If we call `ww_finish` every time, then there's no need for calling `fclose` after `export_wav`, thus making the success boolean unnecessary.

Reply:
These changes would do what you suggest, should I add them to the pull request?
```diff
diff --git a/src/export.c b/src/export.c
index 854ea3c..7454329 100644
--- a/src/export.c
+++ b/src/export.c
@@ -38,8 +38,10 @@ OTHER DEALINGS IN THE SOFTWARE.
 
 extern GfxDomain *domain;
 
-void export_wav(MusSong *song, CydWavetableEntry * entry, FILE *f, int channel)
+bool export_wav(MusSong *song, CydWavetableEntry * entry, FILE *f, int channel)
 {
+	bool success = false;
+	
 	MusEngine mus;
 	CydEngine cyd;
 	
@@ -129,6 +131,8 @@ void export_wav(MusSong *song, CydWavetableEntry * entry, FILE *f, int channel)
 		}
 	}
 	
+	success = true;
+	
 abort:;
 	
 	ww_finish(ww);
@@ -138,5 +142,7 @@ abort:;
 	cyd_deinit(&cyd);
 	
 	song->flags &= ~MUS_NO_REPEAT;
+	
+	return success;
 }
 
diff --git a/src/export.h b/src/export.h
index 8ede779..aa746da 100644
--- a/src/export.h
+++ b/src/export.h
@@ -28,6 +28,6 @@ OTHER DEALINGS IN THE SOFTWARE.
 
 #include "snd/music.h"
 
-void export_wav(MusSong *song, CydWavetableEntry * entry, FILE *f, int channel);
+bool export_wav(MusSong *song, CydWavetableEntry * entry, FILE *f, int channel);
 
 #endif
```

Reply:
Perhaps it's good that it returns the success even if we don't actually use that (yet). :+1: